### PR TITLE
add sample README

### LINKED ISSUE #311: Samples README
We should add Readme for our samples, which explains what each sample has etc.

### PR DESCRIPTION
**Pull request by SaitTalhaNisanci**
This pr adds README for sample folder so that users can see what is in each folder easily.

One concern I have is if any of the files in the links change the landing line will not be showing the correct line in the relevant file.
Fixes #311.

### CODE REVIEW

File: sample/README.md [hunk 1]
```diff
@@ -1,6 +1,64 @@
 <h1>Hazelcast Go Client</h1>
```
Comment (asimarslan) [inline sample/README.md]: Why don't we use markdown format ?

File: sample/README.md [hunk 2]
```diff
@@ -1,6 +1,64 @@
...
+When you have a running Hazelcast instance you will see its address and port in console. You should add this to the **client network config** 
+so that the client can connect to the server. 
+
+```
+config := hazelcast.NewConfig()
+config.NetworkConfig().AddAddress("192.168.1.9:5701") // change this with your address
+
+```
+
+**Don't forget to change the address with your servers address.**
```
Comment (Serdaro) [inline sample/README.md]: **The address given above is an example. Don't forget to change it to to your server's address.**

File: sample/README.md [hunk 3]
```diff
@@ -1,6 +1,64 @@
...
+
 <h2> Examples</h2>
```
Comment (Serdaro) [inline sample/README.md]: Here and below, MarkDown format would be nice.

File: sample/README.md [hunk 4]
```diff
@@ -1,6 +1,61 @@
...
-        "<i>Hello World</i>", the classic entry point to learning
+When you have a running Hazelcast instance, you will see its address and port in the console.  You should add that information to the **network config**, as shown below,
+
+```
+config := hazelcast.NewConfig()
+config.NetworkConfig().AddAddress("192.168.1.9:5701") // change this with your address
+
+```
+
+**The address given above is an example. Don't forget to change it to to your server's address.**
```
Comment (Serdaro) [inline sample/README.md]: ...it to your server's...

File: sample/README.md [hunk 5]
```diff
@@ -1,6 +1,61 @@
...
+```
+config := hazelcast.NewConfig()
+config.NetworkConfig().AddAddress("192.168.1.9:5701") // change this with your address
+
+```
+
+**The address given above is an example. Don't forget to change it to your server's address.**
+## Samples
+
+- ###[flakeidgen](https://github.com/hazelcast/hazelcast-go-client/tree/master/sample/flakeidgen)
```
Comment (asimarslan) [inline sample/README.md]: the hashtags are not needed at here(and below)

Comment (SaitTalhaNisanci) [reply sample/README.md]: @asimarslan  Why are they not needed?

**Pull request merged**